HousingFinanceInterimApi/V1/UseCase/MoveHousingBenefitFileUseCase.cs

```diff
@@ -75,26 +75,25 @@ public async Task<StepResponse> ExecuteAsync()
                 if (!academyFiles.Any())
                     throw new SIO.FileNotFoundException($"No files were found within the '{_academyFileFolderLabel}' label directories.");
 
-                Predicate<File> isValidFileName = (file) => _academyFilePattern.Matches(file.Name).Count == 2;
-
-                var validAcademyFiles = academyFiles.Where(f => isValidFileName(f)).ToList();
-                var notValidAcademyFiles = academyFiles.Where(f => !isValidFileName(f)).ToList();
+                // From the files in the Academy folder, select ones created in last week and rename them.
+                var validRenamedAcademyFiles = academyFiles
+                    .Select(file => new
+                    {
+                        Id = file.Id,
+                        OldName = file.Name,
+                        NewName = CalculateNewFileName(file),
+                        CreatedTime = file.CreatedTime
+                    })
+                    .OrderBy(file => file.CreatedTime)
+                    .ToList();
 
-                foreach (var file in notValidAcademyFiles)
+                var filesCreatedSinceLastWeek = validRenamedAcademyFiles.Where(file => file.CreatedTime > DateTime.Now.AddDays(-7)).ToList();
+                if (!filesCreatedSinceLastWeek.Any())
                 {
-                    var errorMessage = $"Not possible to copy academy files({file.Name})";
-                    LoggingHandler.LogError(errorMessage);
-                    await _batchLogErrorGateway.CreateAsync(batch.Id, "ERROR", $"Application error. {errorMessage}").ConfigureAwait(false);
```

Comment:
Shouldn't the failure to find or process files be reported as a batch log error? From what I recall there is a front-end page somewhere, which gives users the feedback on process completion. Again, I may have missed some refractor talks, so I may not know: Is this no longer a business requirement?

Reply:
This change was part of removing the filename validation, because we no longer depend on it. Any exceptions the UseCase throws will be caught and batch logged though, including no valid files to copy.